invenio_requests/resolvers/requests.py

```diff
@@ -10,49 +10,47 @@
 """Resolver and proxy for requests."""
 
 from invenio_records_resources.references.entity_resolvers import (
-    EntityProxy,
-    EntityResolver,
+    RecordPKProxy,
+    RecordResolver,
 )
-from sqlalchemy.exc import StatementError
-from sqlalchemy.orm.exc import NoResultFound
 
-from ..records.api import Request
+from ..records.api import Request, RequestEvent
+from ..services import RequestEventsServiceConfig, RequestsServiceConfig
 
 
-class RequestProxy(EntityProxy):
-    """Resolver proxy for a Request entity."""
+class RequestResolver(RecordResolver):
+    """Resolver for requests."""
+
+    type_id = "request"
 
-    def _resolve(self):
-        """Resolve the Request from the proxy's reference dict."""
-        request_id = self._parse_ref_dict_id()
-        try:
-            return Request.get_record(request_id)
-        except StatementError as exc:
-            raise NoResultFound() from exc
+    def __init__(self):
+        """Initialize the default record resolver."""
+        super().__init__(
+            record_cls=Request,
+            service_id=RequestsServiceConfig.service_id,
+            type_key=self.type_id,
+            proxy_cls=RecordPKProxy,
+        )
 
-    def get_need(self):
-        """Return None since Needs are not applicable to requests."""
-        return None
+    def _reference_entity(self, entity):
+        """Create a reference dict for the given request."""
+        return {self.type_key: str(entity.id)}
```

Comment:
```suggestion
        id_ = entity.number if entity.number is not None else entity.id
        return {"request": str(id_)}
```

minor: this is the old behaviour for referencing a `Request`, though for some reason, I've never seen a reference dict it like this in practice in our instances... I would keep it maybe just for backwards compatibility.

Reply:
@kpsherva @ntarocco @zzacharo had a quick meeting about this. We did not find any reference in a quick search.
Additionally, the issue is with the `entity.number` being just a counter, starting from 1. When trying to resolve something like this (i.e. `"request": 4`), the record class will not be able to do so, since it is expecting a UUID value.

Reply:
Ok, that makes a lot of sense now. I thought there was some sort of `Request.number.resolve(...)` in place. In that case, we leave it as is :)